### DISCUSSION TITLE
Purpose of num_test argument in TimePDE

### DISCUSSION THREAD

**Discussion by sibyjackgrove** [lululxvi/deepxde#750] [Q&A] [answered]
Could you tell me how the `num_test  `argument is used in `TimePDE`? It seems even if I set it to 1 and call `data.test_x.` I still get a large number of test points.

**Comment by lululxvi** [comment, marked as answer]
This is the testing points for PDE loss, not including BC points.

**Reply by sibyjackgrove** [reply]
@lululxvi  Thank you for the clarification. Is the testing points a subset of the training points? It seems the testing loss is always the same as the training loss. Is there a way to give a set of points that are not part of the training points?

**Reply by lululxvi** [reply]
See the doc at https://deepxde.readthedocs.io/en/latest/modules/deepxde.data.html#module-deepxde.data.pde

**Reply by sibyjackgrove** [reply]
I see from the docs that testing points are randomly sampled from the domain and so should be different from the training points. But what I don't understand is why the training loss is equal to the testing loss?

**Reply by lululxvi** [reply]
"If None, then the training points will be used for testing."

**Reply by sibyjackgrove** [reply]
So if `num_test ` is assigned a non-zero integer, then the testing loss should be different from training loss?

**Reply by lululxvi** [reply]
Yes.

**Reply by sibyjackgrove** [reply]
Ok, thanks! I get it now.
